AppCenterAnalytics/AppCenterAnalytics/Internals/Model/MSEventLog.m

```diff
@@ -98,11 +98,46 @@ - (void)setPropertiesAndMetadataForCSLog:(MSCommonSchemaLog *)csLog {
   if (self.typedProperties) {
     csProperties = [NSMutableDictionary new];
     metadata = [NSMutableDictionary new];
+    NSString *baseDataPrefix = [NSString stringWithFormat:@"%@.", kMSDataBaseData];
+
+    // If there is no valid "baseType" property, there must not be "baseData.*" properties.
+    if (![self.typedProperties.properties[kMSDataBaseType] isKindOfClass:[MSStringTypedProperty class]]) {
+      BOOL removedBaseData = NO;
+      for (NSString *key in [self.typedProperties.properties allKeys]) {
+        if ([key hasPrefix:baseDataPrefix]) {
+          [self.typedProperties.properties removeObjectForKey:key];
+          removedBaseData = YES;
+        }
+      }
+      if (removedBaseData) {
+        MSLogWarning([MSAnalytics logTag], @"baseData was set but baseType is missing or invalid.");
+      }
+
+      // Base type might be set but invalid, so remove it.
+      [self.typedProperties.properties removeObjectForKey:kMSDataBaseType];
+    }
+
+    // If there are no "baseData.*" properties, there must not be a "baseType" property.
```

Comment:
nit: I prefer not to split if/else clauses this way, I would prefer for the comment to be just inside the else block

Reply:
That's not what we do usually and the comment to apply to the block. It should be same level as the other comment.